### PR TITLE
[#84] Add more ignore rules

### PR DESCRIPTION
**Pull request by elbrujohalcon**
[#84] This PR adds ignore rules for `unused_record_fields` and removes the need for ignore rules for the hrl-rules since those work on a file-by-file basis.

### CODE REVIEW

File: src/rules/unused_record_fields.erl [hunk 1]
```diff
@@ -131,7 +131,9 @@ find_record_field(FieldName, Definitions) ->
 
 %% @todo Add ignore pattern support
 -spec ignored(hank_rule:ignore_pattern(), term()) -> boolean().
-ignored(undefined, _IgnoreSpec) ->
-    false; %% Remove this clause and just use the one below
-ignored(_Pattern, _IgnoreSpec) ->
-    true.
+ignored({RecordName, FieldName}, {RecordName, FieldName}) ->
+    true;
+ignored({RecordName, _FieldName}, RecordName) ->
```
Comment (pbrudnick) [inline src/rules/unused_record_fields.erl:136]: I think this case is not tested

Comment (elbrujohalcon) [reply src/rules/unused_record_fields.erl:136]: It is. With `ignored_record`.

Comment (elbrujohalcon) [reply src/rules/unused_record_fields.erl:136]: Here: https://github.com/AdRoll/rebar3_hank/pull/104/files#diff-e7a3faa393edcba8b801bc08843eade4181562b7d76ebfa841e5ef70921fdc81R6

Comment (pbrudnick) [reply src/rules/unused_record_fields.erl:136]: Yes, for this clause:
```erlang
ignored({RecordName, FieldName}, {RecordName, FieldName}) ->
```
but for this:
```erlang
ignored({RecordName, _FieldName}, RecordName) ->
```
I think it's missing something like:
```
-hank([{unused_record_fields,
          [ignored_record,
           {a_record, ignored_field_1},
           {a_record, ignored_field_2},
           {another_complete_record}
           ]}]).
```

File: src/rules/unused_record_fields.erl [hunk 2]
```diff
@@ -131,7 +131,9 @@ find_record_field(FieldName, Definitions) ->
 
 %% @todo Add ignore pattern support
```
Comment (diegomanuel) [inline src/rules/unused_record_fields.erl:132]: The TODO can be removed now! :wink:

File: src/rules/unused_record_fields.erl (hunk 1 above)
Comment (diegomanuel) [reply src/rules/unused_record_fields.erl:136]: The whole `ignored_record` is ignored as Brujo pointed out, so it's covered.

**Review by diegomanuel — APPROVED** [review]
LGTM! :tada:

File: src/rules/unused_record_fields.erl (hunk 1 above)
Comment (pbrudnick) [reply src/rules/unused_record_fields.erl:136]: Yes, sorry, I got confused between the rulename and the recordname

**Review by pbrudnick — APPROVED** [review]

**Pull request merged**